compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -781,6 +791,7 @@ function applyEffect(
       const fromValue = state.kind(effect.from);
       const fromKind = fromValue.kind;
       switch (fromKind) {
+        case ValueKind.ShallowMutable:
```

Comment:
hmm this doesn't seem right, we should keep this as an Assign. The new rule is:

```
Given
  x is ShallowMutable
  CreateFrom y <- x
Then
  y is Frozen
```

Whereas this basically says

```
Given
  x is ShallowMutable
  Assign y = x
Then
  y is Frozen (bc immutable capture)
```